lib/Extractor/KLEEBuilder.cpp

```diff
@@ -366,6 +366,50 @@ class KLEEBuilder : public ExprBuilder {
       unsigned Index = Ops[1]->Val.getZExtValue();
       return get(Ops[0]->Ops[Index]);
     }
+    case Inst::SAddSat:
+    {
+      ref<Expr> add = AddExpr::create(get(Ops[0]), get(Ops[1]));
+      auto sextL = SExtExpr::create(get(Ops[0]), I->Width + 1);
+      auto sextR = SExtExpr::create(get(Ops[1]), I->Width + 1);
+      auto addExt = AddExpr::create(sextL, sextR);
+      auto smin = klee::ConstantExpr::alloc(llvm::APInt::getSignedMinValue(I->Width));
+      auto smax = klee::ConstantExpr::alloc(llvm::APInt::getSignedMaxValue(I->Width));
+      auto sminExt = SExtExpr::create(smin, I->Width + 1);
+      auto smaxExt = SExtExpr::create(smax, I->Width + 1);
+      auto pred = SleExpr::create(addExt, sminExt);
+
+      auto pred2 = SgeExpr::create(addExt, smaxExt);
+      auto select2 = SelectExpr::create(pred2, smax, add);
+
+      return SelectExpr::create(pred, smin, select2);
+    }
+    case Inst::UAddSat:
+    {
```

Comment:
prefer putting this kind of curly brace on the same line as the case
(don't worry about it for this PR but do it generally)